### PR TITLE
CLOUDP-230380: Improve Release documentation

### PR DESCRIPTION
**Pull request by josvazg**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

File: docs/dev/release.md [hunk 1]
```diff
@@ -1,34 +1,29 @@
...
 
-Review the Pull Request. Approve and merge it to main.
+Review the Pull Request. Approve and merge it to `main`.
 The new job "Create Release" will be triggered and the following will be done:
-* Atlas Operator image built and pushed to dockerhub
+* Atlas Operator image is built and pushed to DockerHub
 * Draft Release will be created with all commits since the previous release
 
-Once the Pull Request is approved, a tag is created out of the branch, which can then be discarded. A branch `release/X.Y.Z` will imply a tag `vX.Y.Z`, and `pre-release/X.Y.Z-...` will imply `vX.Y.Z-...`. The `tag.yml` workflow is the one responsible for creating such tag and triggering the release process workflow (`release-post-merge.yml`).
+Once the Pull Request is approved, a tag is created out of the branch, so such branch could be discarded. A branch `release/X.Y.Z` turns into tag `vX.Y.Z`, and `pre-release/X.Y.Z-...` into `vX.Y.Z-...`. The `tag.yml` workflow is the one responsible for creating a tag from the branch closed, and triggering the release process workflow (`release-post-merge.yml`).
```
Comment (s-urbaniak) [inline docs/dev/release.md:22]:
```suggestion
The "Create Release Branch" workflow is going to create a Pull Request pointing to a `release/X.Y.Z` branch. Once approved and merged, automation is going to create a `vX.Y.Z` tag.
```
I wouldn't mention `tag.yml`. It confuses the person who is doing the release. The statement:
```
The `tag.yml` workflow is the one responsible for creating a tag from the branch closed, and triggering the release process workflow (`release-post-merge.yml`).
```
is rather suited to be placed in the separate markdown which is documenting GitHub actions.

File: docs/dev/release.md [hunk 2]
```diff
@@ -1,34 +1,29 @@
...
 
 ## Edit the Release Notes and publish the release
 
-Follow the format described in the release-notes-template.md file. Publish the release.
+Follow the format described in the [release-notes-template.md](../release-notes/release-notes-template.md) file. Publish the release.
```
Comment (s-urbaniak) [inline docs/dev/release.md:26]:
```suggestion
Follow the format described in the [release-notes-template.md](../release-notes/release-notes-template.md) file. Before publishing the release, keep the release in Draft and get an approval from the team and Product Management.
```

File: docs/dev/release.md [hunk 3]
```diff
@@ -49,7 +44,12 @@ All bundles/package manifests for Operators for operatorhub.io reside in the fol
 
 ### Fork/Update the community operators repositories
 
-**Note**: this has to be done once only:
+**Note**: this has to be done once only. 
+
+First ensure your SSH keys in [https://github.com/settings/keys] are authorized for `mongodb-forks` MongoDB SSO.
+
+Then, from your usual directory holding your git repository copies, do the following:
```
Comment (s-urbaniak) [inline docs/dev/release.md:51]:
```suggestion
Execute the following steps:
```

File: docs/dev/release.md [hunk 4]
```diff
@@ -73,6 +73,12 @@ git remote add upstream https://github.com/redhat-openshift-ecosystem/certified-
 export RH_CERTIFIED_OPENSHIFT_REPO_PATH=$PWD/certified-operators
 ```
 
+### Before creating PRs against RedHat repositories
```
Comment (s-urbaniak) [inline docs/dev/release.md:76]: This should be in a `### Prerequisites` section at the beginning of the document.

File: docs/dev/release.md [hunk 5]
```diff
@@ -73,6 +73,12 @@ git remote add upstream https://github.com/redhat-openshift-ecosystem/certified-
...
+
+To get PRs to be auto-committed for RedHat community & Openshift you need to make sure you are listed in the [team members ci.yaml list (community-operators)](https://github.com/k8s-operatorhub/community-operators/blob/main/operators/mongodb-atlas-kubernetes/ci.yaml). Link for [community-operators-prod](https://github.com/redhat-openshift-ecosystem/community-operators-prod/blob/main/operators/mongodb-atlas-kubernetes/ci.yaml). This is not a requirement for [Certified Operators](https://github.com/redhat-openshift-ecosystem/certified-operators/blob/main/operators/mongodb-atlas-kubernetes/ci.yaml).
```
Comment (s-urbaniak) [inline docs/dev/release.md:78]:
```suggestion
To get PRs to be auto-committed for RedHat community & Openshift you need to make sure you are listed in the [team members ci.yaml list for community-operators](https://github.com/k8s-operatorhub/community-operators/blob/main/operators/mongodb-atlas-kubernetes/ci.yaml) and [team members ci.yaml list for community-operators-prod](https://github.com/redhat-openshift-ecosystem/community-operators-prod/blob/main/operators/mongodb-atlas-kubernetes/ci.yaml).

This is not required for [Certified Operators](https://github.com/redhat-openshift-ecosystem/certified-operators/blob/main/operators/mongodb-atlas-kubernetes/ci.yaml).
```

File: docs/dev/release.md [hunk 6]
```diff
@@ -73,6 +73,12 @@ git remote add upstream https://github.com/redhat-openshift-ecosystem/certified-
...
+
+All this is in addition to holding a RedHat Connect account adn being a [team member with org administrator role in the team list](https://connect.redhat.com/account/team-members), which is part of the *team onboarding*.
```
Comment (s-urbaniak) [inline docs/dev/release.md:80]:
```suggestion
Finally, make sure you have a "RedHat Connect" account and are a [team member with org administrator role in the team list](https://connect.redhat.com/account/team-members).
```

File: docs/dev/release.md [hunk 7]
```diff
@@ -145,6 +151,30 @@ You can see an [example fixed PR here for certified version 1.9.1](https://githu
 
 After the PR is approved it will soon appear in the [Atlas Operator openshift cluster](https://console-openshift-console.apps.atlas.operator.mongokubernetes.com)
 
+# SSDLC checklist publishing
+
+For the time being, preparing the SSDLC checklist for each release is a manual process. Use this [past PR as a starting point](https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1524).
+
+Copy the closest [sdlc-compliance.md](../releases/v2.2.1/sdlc-compliance.md) file and:
+- Update the **version** references to the one being released.
+- Update dates and release creators to match the reality of the current release.
```
Comment (s-urbaniak) [inline docs/dev/release.md:160]:
```suggestion
- Update dates and release creators of the current release.
```

File: docs/dev/release.md [hunk 8]
```diff
@@ -145,6 +151,30 @@ You can see an [example fixed PR here for certified version 1.9.1](https://githu
...
+- Usually no more changes will be needed. Only if you actually skipped some CI check, like allowed the release to happen even when some report flagged a dependency that had no fix yet, you might need to mention such case.
```
Comment (s-urbaniak) [inline docs/dev/release.md:161]: This sentence is confusing for the reader. Let's shorten this to the minimum or leave out.

File: docs/dev/release.md [hunk 9]
```diff
@@ -145,6 +151,30 @@ You can see an [example fixed PR here for certified version 1.9.1](https://githu
...
+
+There is no need to make any changes about the image signature part other than updating the instructions to the current version.
```
Comment (s-urbaniak) [inline docs/dev/release.md:163]: What does that mean concretely?
```suggestion
Update the image signature instructions to match the current version.
```

File: docs/dev/release.md [hunk 10]
```diff
@@ -145,6 +151,30 @@ You can see an [example fixed PR here for certified version 1.9.1](https://githu
...
+
+For SBOMs, you will have to *generate* the files and place them in the same directory as the compliance doc `docs/releases/vX.Y.Z`, with the expected names; `linux-amd64.sbom.json` & `linux-arm64.sbom.json`. To generate each f the files you need to run:
```
Comment (s-urbaniak) [inline docs/dev/release.md:165]:
```suggestion
Generate the `linux-amd64.sbom.json` and `linux-arm64.sbom.json` SBOM files and place them in the same directory as the compliance doc `docs/releases/vX.Y.Z`:
```

File: docs/dev/release.md [hunk 11]
```diff
@@ -145,6 +151,30 @@ You can see an [example fixed PR here for certified version 1.9.1](https://githu
...
+```shell
+docker sbom --platform "linux/${arch}" -o "docs/releases/v${version}/linux-${arch}.sbom.json" --format "cyclonedx-json" "$image"
+```
+
+Where:
+- `${arch}` is the architecture to generate, either `amd64` or `arm64`.
+- `${version}` is the current version released in `X.Y.Z` format, without the **v** prefix.
+- `${image}` is the image reference released, usually something like `mongodb/mongodb-atlas-kubernetes-operator:${version}`.
+
+Once all such information is in place, create a PR with it and merge it as close to the release as possible.
```
Comment (s-urbaniak) [inline docs/dev/release.md:176]:
```suggestion
Create a PR with the following new files included in the `releases/vX.Y.Z` directory:
- `linux-amd64.sbom.json`
- `linux-arm64.sbom.json`
- `sdlc-compliance.md`
```

File: docs/dev/release.md [hunk 12]
```diff
@@ -171,3 +201,15 @@ git push
 ```
 
 If the release is a new minor version, then the CLI must be updated with the new version (and any new CRDs) [here](https://github.com/mongodb/mongodb-atlas-cli/blob/master/internal/kubernetes/operator/features/crds.go).
+
+## Troubleshooting
+
+### Major version issues when Create Release Branch
```
Comment (s-urbaniak) [inline docs/dev/release.md:207]:
```suggestion
### Major version issues when executing the "Create Release Branch" workflow
```

**Review by s-urbaniak — CHANGES_REQUESTED** [review]
just couple of suggestion, thank you for the updated release notes 👍

File: docs/dev/release.md (hunk 8 above)
Comment (josvazg) [reply docs/dev/release.md:161]: OK
```suggestion
```

File: docs/dev/release.md (hunk 4 above)
Comment (josvazg) [reply docs/dev/release.md]: Moved

File: docs/dev/release.md (hunk 8 above)
Comment (s-urbaniak) [reply docs/dev/release.md]: ok, lgtm

**Review by s-urbaniak — APPROVED** [review]
